src/java/org/apache/cassandra/service/CassandraDaemon.java

```diff
@@ -291,7 +300,7 @@ protected void setup()
         {
             SystemKeyspace.snapshotOnVersionChange();
         }
-        catch (IOException e)
+        catch (Throwable e)
```

Comment:
why the change here?

Reply:
because what it internally does is that it takes a snapshot and `executeTask` in snapshot manager will ultimately throw SnapshotException which is RuntimeException, not IOException.